src/IECoreMaya/ToMayaPlugConverter.cpp

```diff
@@ -68,12 +71,95 @@ ToMayaPlugConverterPtr ToMayaPlugConverter::create( const IECore::ObjectPtr src
 	return new ToMayaPlugConverter( src );
 }
 
-bool ToMayaPlugConverter::convert( MPlug &plug ) const
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::setPlugValue(MPlug &plug, typename IECore::TypedData<DataType>::ConstPtr data)
 {
-	ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+	return plug.setValue(data->readable());
+}
+
+
+// specializations for Unit Attributes
+namespace IECoreMaya
+{
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MTime>(MPlug &plug, typename IECore::TypedData<float>::ConstPtr data)
+{
+	MTime t;
+	t.setUnit(MTime::kSeconds);
+	t.setValue(data->readable());
+	return plug.setValue(t);
+}
 
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MDistance>(MPlug &plug, typename IECore::TypedData<float>::ConstPtr data)
+{
+	MDistance t;
+	t.setUnit(MDistance::kCentimeters);
+	t.setValue(data->readable());
+	return plug.setValue(t);
+}
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MAngle>(MPlug &plug, typename IECore::TypedData<float>::ConstPtr data)
+{
+	MAngle t;
+	t.setUnit(MAngle::kRadians);
+	t.setValue(data->readable());
+	return plug.setValue(t);
+}
+
+} // namespace IECoreMaya
+
+
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::convertAttr(MPlug &plug) const
+{
 	MStatus s;
+	if (plug.isArray())
+	{
+		ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+		IECore::CompoundData::ConstPtr data = IECore::runTimeCast<const IECore::CompoundData>( toConvert );
```

Comment:
Like the non-array case, we can simplify this a bit

`IECore::CompoundData::ConstPtr data = IECore::runTimeCast<const IECore::CompoundData>( srcParameter()->getValidatedValue() );`